Document FERC SQLite Provenance; Update RTD docs links (#5222)
catalyst-cooperative/pudl: docs/dev/clone_ferc1.rst

choose to materialize any combination of form an format (DBF or XBRL). Note that the
earlier FERC 714 data (2006-2020) is distributed as CSVs, and PUDL does not yet load
all available tables.

FERC SQLite provenance compatibility
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

PUDL records provenance metadata when FERC SQLite assets are materialized and checks
that metadata before downstream ``pudl`` assets use those databases. This helps avoid
scenarios where local databases were built with different inputs or settings than the

> current run.

Reviewer: I'm having trouble tracking clauses when everything is squeezed into two sentences. maybe:

```suggestion
PUDL records provenance metadata when FERC SQLite assets are materialized, including
the Zenodo DOI and the years extracted. 
Before downstream ``pudl`` assets use those databases, 
PUDL checks the metadata to make sure they are compatible with the current run.
This helps avoid scenarios where the FERC databases were built with old inputs
or don't have all the years needed by the current run.
```

Incompatible provenance metadata
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The provenance check looks at two things:

* Whether the Zenodo DOI changed for a FERC dataset. This would  mean that the expected

>   raw input data is different from what was used to build the existing SQLite DB.

Reviewer: ```suggestion
* Whether the Zenodo DOI changed for a FERC dataset. This would mean that the
  raw input data used to build the existing SQLite DB is different from what is expected.
```

* Whether the current ETL config requests years of FERC data that are not present in the
  existing SQLite DB.

When either of these criteria aren't met, Dagster raises an error telling you to refresh
the FERC SQLite assets.

> Regenerating FERC SQLite prerequisites

Reviewer: are these prerequisites for FERC SQLite or prerequisites for PUDL? I think we mean the same thing we meant by "ferc sqlite assets" above so maybe we can use that to be unambiguous:

```suggestion
Regenerating FERC SQLite assets
```
